File: leetcode/solutions/1281.SubtractTheProductAndSumOfDigitsOfAnInteger/subtractProductAndSum_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import subtractProductAndSum from "./subtractProductAndSum.ts";

Deno.test("1281. Subtract the Product and Sum of Digits of an Integer", () => {
  type test = [number, number];
  const tests: test[] = [
    [0, 0],
    [1, 0],
    [999999, 531387],
    [999990, -45],
    [4420, -10],
    [234, 15],
    [4421, 21],
  ];

  for (const [arg, want] of tests) {
    assertEquals(subtractProductAndSum(arg), want);
  }
});

File: leetcode/solutions/1281.SubtractTheProductAndSumOfDigitsOfAnInteger/subtractProductAndSum_test.go
package solution

import (
	"testing"
)

func Test_1281_SubtractTheProductAndSumOfDigitsOfAnInteger(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  int
		out int
	}{
		{0, 0},
		{1, 0},
		{999999, 531387},
		{999990, -45},
		{4420, -10},
		{234, 15},
		{4421, 21},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := subtractProductAndSum(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
